### PR TITLE
Wrap ActiveRecord::RecordNotUnique with Sequent::Core::EventStore::OptimisticLockingError

### PR DESCRIPTION
**Pull request by derekkraan**
This will impact your app if you are handling `ActiveRecord::RecordNotUnique` originating from the event store.

@bjpbakker @s0meone @erikrozendaal comments?

### CODE REVIEW

**Comment by lvonk** [comment]
Ping @bjpbakker @s0meone. We want to wrap the RecordNotUnique so we can react to this. If you are using this somehow you will need to change this :-)

**Comment by bjpbakker** [comment]
:+1:

File: lib/sequent/core/event_store.rb [hunk 1]
```diff
@@ -164,11 +172,14 @@ def store_events(command, streams_with_events = [])
                       created_at: event.created_at}
             values = values.merge(organization_id: event.organization_id) if event.respond_to?(:organization_id)
 
-            event_record_class.create!(values)
+            begin
+              event_record_class.create!(values)
+            rescue ActiveRecord::RecordNotUnique => e
```
Comment (bjpbakker) [inline lib/sequent/core/event_store.rb]: `e` is unused

Comment (derekkraan) [inline lib/sequent/core/event_store.rb]: thanks, fixed!

**Pull request merged**